flexmeasures/ui/templates/crud/asset_graph.html

```diff
@@ -0,0 +1,746 @@
+        // Due to the nature of async functions, the highlightedCard might not be available
+        // when the filterSensors function is called. So, we need to check if it exists
+        if (highlightedCard) {
+            const cardId = highlightedCard.id;
+            const index = cardId.split("_")[1];
+            savedGraphIndex = index;
+            selectedGraphTitle = sensorsToShow[index].title;
+        } else {
+            savedGraphIndex = undefined;
+            selectedGraphTitle = undefined;
+        }
+
+        // check if apiSensorsList has been rendered
+        if (apiSensorsListElement.innerHTML === "") {
+            document.getElementById('apiSensorsList').style.display = 'block';
+        }
+
+        const params = new URLSearchParams();
+
+        if (searchValue) {
+            params.append('filter', searchValue);
+        }
+
+        if (filterValue !== "Units") {
+            params.append('unit', filterValue);
+        }
+
+        const apiUrl = `${sensorsApiUrl}&${params.toString()} `;
+
+        try {
+            const response = await fetch(apiUrl);
+
+            if (!response.ok) {
+                throw new Error('Failed to fetch sensors');
+            }
+
+            const responseData = await response.json();
+            const filteredSensors = responseData.data;
+
+            // Render the fetched sensors
+            if (savedGraphIndex !== null && savedGraphIndex !== undefined) {
+                renderApiSensors(filteredSensors, savedGraphIndex);
+            } else {
+                renderApiSensors(filteredSensors);
+            }
+
+            cachedFilteredSensors = filteredSensors;
+
+            spinnerElement.classList.add('hidden-important');
+            apiSensorsListElement.style.display = 'block';
+        } catch (error) {
+            showToast("Failed to filter/search sesnors", "error");
+            console.error(error);
+        }
+    }
+
+    async function searchSensors() {
+        const searchValue = document.getElementById("flexContextSensorSearch").value.toLowerCase();
+        spinnerElement.style.display = 'flex';
+        senSearchResEle.style.display = 'none';
+
+        const params = new URLSearchParams();
+
+        if (searchValue) {
+            params.append('filter', searchValue);
+        }
+
+        const apiUrl = `${sensorsApiUrl}&${params.toString()} `;
+
+        try {
+            const response = await fetch(apiUrl);
+
+            if (!response.ok) {
+                throw new Error('Failed to fetch sensors');
+            }
+
+            const responseData = await response.json();
+            const filteredSensors = responseData.data;
+
+            // Render the fetched sensors
+            renderSensorSearchResults(filteredSensors);
+
+            spinnerElement.classList.add('hidden-important');
+            senSearchResEle.style.display = 'block';
+        } catch (error) {
+            showToast("Failed to filter/search sesnors", "error");
```

Comment:
Huh, there is a pre-existing typo: "sesnors". Can you fix it? It appears twice, I'll also comment on the other one.

Also, we can just say "search" here.